text/0030-genesis-validator-lockup.md

```diff
@@ -0,0 +1,57 @@
+# Lock-up of genesis validators and access to tokens staked at genesis
+
+## Summary
+
+[summary]: #summary
+
+CEP PR: [casperlabs/ceps#0030](https://github.com/casperlabs/ceps/pull/0030)
+
+We propose that the operation of the auction and related contracts be modified to reflect the language of agreements regarding token allocations. To this end, we propose that the unlock of tokens be gradual and take place over the course of 90 days. Further, we introduce a new type of deploy, the genesis validator token withdrawal, which would enable genesis validators to withdraw eligible tokens without paying for gas.
+
+## Motivation
+
+[motivation]: #motivation
+
+Tokens held by genesis validators are subject to legal agreements that specify an unlock period. Presently, the auction contract only implements a "cliff" at 90 days, but the unlock is expected to take place over the following 90 days, rather than instantly. Additionally, this proposal aims to resolve the expected shoftfall of circulating tokens at launch, as all amounts allocated to genesis validators will be staked. To this end, we propose introduction of a new time-limited deploy type specifically for reward withdrawals.
+
+## Guide-level explanation
+
+[guide-level-explanation]: #guide-level-explanation
+
+We introduce two changes to the platform. The first change concerns the lock-in period and auto-win conditions managed by the auction contract:
+
+- Total lock-in for 90 days remains in place
+- After 90 days, tokens unlock gradually (equal proportion of starting amount) over the course of a further 90 days, possibly as weekly lump sums
+- Auto-win conditions is set to false at the 90 day mark and genesis validators begin to compete in the auction
+
+The second change concerns the problem of genesis validators having no tokens to withdraw their rewards. To this end, we propose introducing a new deploy type, the genesis validator token withdrawal, which requires no payment for gas and can be processed by the platform for the first **210 days**.
```

Comment:
We expect people to stake millions of dollars and not expect them to buy enough tokens to make the withdrawal (which should be a couple of cents)?

Reply:
There are no tokens to buy at the beginning of time, since the initial validators' tokens are locked into stakes. The first liquidity comes from them withdrawing their rewards and unlocked portions of the stake.

Reply:
There are people, other than validators, whose public keys are in the initial `accounts.csv` file – they don't have any tokens staked but they had bought tokens on pre-sales and their initial balance is non-zero.

Reply:
The account.csv could be updated so that for each genesis validator, the value in the initial balance column is enough to initiate withdrawal and the initial staked amount is reduced accordingly.

Reply:
@goral09, do you mean genesis delegators, perhaps? To date, all information I saw suggested that having to stake tokens "circulating" at genesis was a contractual obligation that actually made the sales possible. It's true that there is a source of liquidity from post-launch sales to non-validators, of course, but it's not ideal to have validators wait for those before being able to access rewards. (sorry, I meant to reply earlier, but events intervened)

@momipsl, I think I get the gist of what you're suggesting, but I thought having genesis validators get unbonded tokens was contrary to the token sale agreements